src/rgw/rgw_civetweb_frontend.cc

```diff
@@ -64,6 +64,9 @@ int RGWCivetWebFrontend::run()
     conf_map["run_as_user"] = std::move(uid_string);
   }
 
+  /* Set authentication domains, this is needed for handling absolute uris in
+     requests */
+  conf_map["authentication_domain"] = g_conf->rgw_dns_name;
```

Comment:
Yeah, so we should modify civetweb (add a new configurable to allow that so that it can be accepted upstream).